src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -1,652 +1,399 @@
-        }
-      },
-    );
-  }
-
-  afterInit(entityForm: EntityFormComponent): void {
-    const generalFieldsets = _.find(this.fieldSets, { class: 'basic' });
-    const purposeField = _.find(generalFieldsets.config, { name: 'purpose' }) as FormSelectConfig;
-    this.ws.call('sharing.smb.presets').pipe(untilDestroyed(this)).subscribe(
-      (presets) => {
-        this.presets = presets;
-        for (const item in presets) {
-          purposeField.options.push({ label: presets[item]['verbose_name'], value: item });
+  ngOnInit(): void {
+    this.form.get('purpose').valueChanges.pipe(untilDestroyed(this)).subscribe(
+      (value: string) => {
+        this.clearPresets();
+        if (!this.presets[value]) {
+          return;
         }
-        if (entityForm.isNew) {
-          entityForm.formGroup.controls['purpose'].setValue('DEFAULT_SHARE');
+        for (const param in this.presets[value].params) {
+          this.presetFields.push(param as keyof SmbShare);
+          const ctrl = this.form.get(param);
+          if (ctrl && param !== 'auxsmbconf') {
+            ctrl.setValue(this.presets[value].params[param as keyof SmbShare]);
+            ctrl.disable();
+          }
         }
       },
-      (err) => {
-        new EntityUtils().handleWsError(this, err, this.dialog);
-      },
     );
 
-    this.entityForm = entityForm;
-    this.title = entityForm.isNew ? helptextSharingSmb.formTitleAdd : helptextSharingSmb.formTitleEdit;
-    if (entityForm.isNew) {
-      entityForm.formGroup.controls['browsable'].setValue(true);
-    } else {
-      setTimeout(() => {
-        entityForm.formGroup.controls['aapl_name_mangling'].valueChanges.pipe(untilDestroyed(this)).subscribe((value) => {
-          if (value !== this.mangle && !this.mangleWarningSent) {
-            this.mangleWarningSent = true;
-            this.dialog.confirm({
-              title: helptextSharingSmb.manglingDialog.title,
-              message: helptextSharingSmb.manglingDialog.message,
-              hideCheckBox: true,
-              buttonMsg: helptextSharingSmb.manglingDialog.action,
-              hideCancel: true,
-            });
-          }
-        });
-      }, 1000);
-    }
+    this.form.get('afp').valueChanges.pipe(untilDestroyed(this)).subscribe((value: boolean) => {
+      this.afpConfirm(value);
+    });
 
+    const pathFormControl = this.form.get('path');
     /*  If name is empty, auto-populate after path selection */
-    entityForm.formGroup.controls['path'].valueChanges.pipe(untilDestroyed(this)).subscribe((path) => {
-      const nameControl = entityForm.formGroup.controls['name'];
+    pathFormControl.valueChanges.pipe(debounceTime(50), untilDestroyed(this)).subscribe((path) => {
+      const nameControl = this.form.get('name');
       if (path && !nameControl.value) {
         const name = path.split('/').pop();
         nameControl.setValue(name);
       }
 
       if (!this.stripACLWarningSent) {
```

Comment:
1. Bad property name. => `wasStripAclWarningShown` or something.
2. Can return early.
3. Same for res here and in other places.
4. This block and block below look similar. Any way to combine (e.g. with `combineLatest`).